File: src/main/java/de/tectoast/emolga/commands/music/control/PauseCommand.java
package de.tectoast.emolga.commands.music.control;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.commands.MusicCommand;
import de.tectoast.emolga.utils.music.GuildMusicManager;

public class PauseCommand extends MusicCommand {

    public PauseCommand() {
        super("pause", "Pausiert den derzeitigen Track oder setzt ihn fort, wenn er pausiert ist");
        setArgumentTemplate(ArgumentManagerTemplate.noArgs());
    }

    @Override
    public void process(GuildCommandEvent e) {
        GuildMusicManager manager = getGuildAudioPlayer(e.getGuild());
        if (manager.player == null) {
            e.reply("Derzeit läuft kein Track!");
            return;
        }
        AudioPlayer player = manager.player;
        if (player.getPlayingTrack() == null) {
            e.reply("Derzeit läuft kein Track!");
            return;
        }
        player.setPaused(!player.isPaused());
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/music/control/PauseCommand.kt
package de.tectoast.emolga.commands.music.control

import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.commands.MusicCommand

class PauseCommand : MusicCommand("pause", "Pausiert den derzeitigen Track oder setzt ihn fort, wenn er pausiert ist") {
    init {
        argumentTemplate = ArgumentManagerTemplate.noArgs()
    }

    override suspend fun process(e: GuildCommandEvent) {
        val manager = getGuildAudioPlayer(e.guild)
        val player = manager.player
        if (player.playingTrack == null) {
            e.reply("Derzeit läuft kein Track!")
            return
        }
        player.isPaused = !player.isPaused
    }
}